feat: implement GoToStateAction

pymendoza opened the pull request.

barjonp commented:
Hi, have you tried using the original implementation ? [GoToStateAction.cs](https://github.com/microsoft/XamlBehaviorsWpf/blob/2caaab1a869efbd5ebc9a00b187dccca54fec818/Source/Interactions/GoToStateAction.cs)

pymendoza commented:
> Hi, have you tried using the original implementation ? [GoToStateAction.cs](https://github.com/microsoft/XamlBehaviorsWpf/blob/2caaab1a869efbd5ebc9a00b187dccca54fec818/Source/Interactions/GoToStateAction.cs)

Thanks for checking. I just updated the commit to match closer to the original implementation. Please note I extracted the logic that sets the StateTarget property onto a new TrySetStateTarget() method to allow handling configuration where TargetObject dependency property is not explicitly specified like this, which is allowed in SL:

<ei:GoToStateAction StateName="ErrorState" />
